### ISSUE TITLE
[Bug]: Add 'M³' to the media window title

### ISSUE THREAD

**Issue by ux-git** [sircharlo/meeting-media-manager#3668]
### What happened?

Not an issue, but to make the relationship between this window and M³ even more obvious, it's name can start with M³: 'M³ Media Player Window'

### To Reproduce

-

### What did you expect to happen?

-

### Version

v24.12.0

### What platform are you seeing the problem on?

Windows

### Screenshots

![IMG_20241207_184933_949](https://github.com/user-attachments/assets/61a8cdcb-05e3-44f5-85c3-638ee3e2f1a8)

### Confirmations

- [X] I confirm that I have read the documentation website.
- [X] I confirm that I have searched through the existing issues to make sure my bug report is not a duplicate.
- [X] I confirm that I have searched through the existing discussions to make sure my bug report is not a duplicate.

**Comment by mtdvlpr** [comment]
Implemented in 5ee636c5cb9c96be2c3cfab4a39ac6c52c3cb665

**Comment by sircharlo** [comment]
Potential breaking change: some KH use the window name for the OBS share

**Comment by mtdvlpr** [comment]
> Potential breaking change: some KH use the window name for the OBS share

Didn't think of that. Do you want to revert it?

**Comment by fschein** [comment]
> some KH use the window name for the OBS share

We use it for two reasons: it prevents the mouse from accidentally appearing in the streming and it also avoids the infinite window effect if the Zoom window is displayed on the screen.
I don't mind changing that in OBS, but it could be an issue for congregations with less technical knowledge.

**Comment by sircharlo** [comment]
My thoughts exactly. Maybe we should either warn of this change, or make it user-configurable. Otherwise, I'm happy leaving it as is too.

**Comment by mtdvlpr** [comment]
> My thoughts exactly. Maybe we should either warn of this change, or make it user-configurable. Otherwise, I'm happy leaving it as is too. 

We could use the announcements feature for warning the users.

**Comment by fschein** [comment]
With the announcement it will definitely be resolved! And I think it's a great idea to better describe the window name.

**Comment by mtdvlpr** [comment]
This will be released in `v25.1.0`. We'll add the announcement first to give users a heads up.

**Comment by fschein** [comment]
How does this notification work? For us, it came in English, and there's no string for it on Crowdin. Is it possible to translate it?

**Comment by mtdvlpr** [comment]
> How does this notification work? For us, it came in English, and there's no string for it on Crowdin. Is it possible to translate it?

The announcements are retrieved directly from GitHub. That's why you're seeing it without an app update. We can make it translatable, but the translations would require an app update, so they won't appear until a new release with the translations included is installed. That's why I figured it wouldn't make sense to make this temporary announcement translatable.

**Comment by mtdvlpr** [comment]
I've disables the announcement again until the next release, since it was now showing for every user instead of just OBS users. The next release included a feature to only show announcements to a certain scope of users.

**Comment by sircharlo** [comment]
> > How does this notification work? For us, it came in English, and there's no string for it on Crowdin. Is it possible to translate it?
> 
> The announcements are retrieved directly from GitHub. That's why you're seeing it without an app update. We can make it translatable, but the translations would require an app update, so they won't appear until a new release with the translations included is installed. That's why I figured it wouldn't make sense to make this temporary announcement translatable.

🤔 It might be complicated, but I feel that we could have a folder for announcement translations that gets synced with Crowdin and downloads the translated strings from the GitHub API.

**Comment by bumblebeelabs** [comment]
We are using a modified Zoom Workplace (Zoom Meeting SDK). We have a button to restore M³'s main window in case it gets lost. When M³ auto updated to 24.12.1 and its title was changed to "Meeting Media - M³", we had a little shock.

The change to "M³ Media Player Window" will have a greater impact, since we programmatically restore "Media Player Window" after stopping spotlighting.

🤔 Will this change to M³ window titles be final, or should we expect M³ window titles to change from time to time? Or is there another constant we can look for to uniquely identify each of M³'s windows?

**Comment by mtdvlpr** [comment]
> We are using a modified Zoom Workplace (Zoom Meeting SDK). We have a button to restore M³'s main window in case it gets lost. When M³ auto updated to 24.12.1 and its title was changed to "Meeting Media - M³", we had a little shock.
> 
> The change to "M³ Media Player Window" will have a greater impact, since we programmatically restore "Media Player Window" after stopping spotlighting.
> 
> 🤔 Will this change to M³ window titles be final, or should we expect M³ window titles to change from time to time? Or is there another constant we can look for to uniquely identify each of M³'s windows?

Sorry to hear that! We knew the media window title change could cause breaking changes, but I didn't consider your use case. The title changes are final. Although, thinking about your use case, it might still be broken when you're using the "Present website" feature, since that has a different window title than that of the "Meeting media" page. Would you say your setup requires each page (other than the media window) should have the same window title?

**Comment by bumblebeelabs** [comment]
> Sorry to hear that! We knew the media window title change could cause breaking changes, but I didn't consider your use case. The title changes are final. Although, thinking about your use case, it might still be broken when you're using the "Present website" feature, since that has a different window title than that of the "Meeting media" page. Would you say your setup requires each page (other than the media window) should have the same window title?

You are quite right about that, and on top of that, there's the effect of internationalization. May I confirm, from [MainLayout.vue](https://github.com/sircharlo/meeting-media-manager/blob/dfa00afb4f851daeeb16bd06e181051191ff129d/src/layouts/MainLayout.vue#L89), to find the main window, the window title must match `.\+ - M³\|Meeting Media Manager`?

And to find the media player window, from [MediaPlayerLayout.vue](https://github.com/sircharlo/meeting-media-manager/blob/dfa00afb4f851daeeb16bd06e181051191ff129d/src/layouts/MediaPlayerLayout.vue#L15), the string to match in the future is confirmed to be `M³ Media Player Window`?

**Comment by mtdvlpr** [comment]
You are correct about the main window title. The media player window title will become `Media Player - M³`
